File: datasets/rb-keep-code_davinci_001_temp_0.2/HumanEval_79_decimal_to_binary.rb
# You will be given a number in decimal form and your task is to convert it to
# binary format. The function should return a string, with each character representing a binary
# number. Each character in the string will be '0' or '1'.
# There will be an extra couple of characters 'db' at the beginning and at the end of the string.
# The extra characters are there to help with the format.
# Examples:
# decimal_to_binary(15)   # returns "db1111db"
# decimal_to_binary(32)   # returns "db100000db"
def decimal_to_binary(decimal)
  binary = []
  while decimal > 0
    binary.unshift(decimal % 2)
    decimal /= 2
  end
  "db#{binary.join}db"
end

require 'test/unit'
class TestHumanEval < Test::Unit::TestCase
  def test_decimal_to_binary
    candidate = method(:decimal_to_binary)
    assert_equal("db0db", candidate.call(0))
    assert_equal("db100000db", candidate.call(32))
    assert_equal("db1100111db", candidate.call(103))
    assert_equal("db1111db", candidate.call(15))
  end
end

File: datasets/r-keep-code_davinci_001_temp_0.2/HumanEval_79_decimal_to_binary.r
# You will be given a number in decimal form and your task is to convert it to
# binary format. The function should return a string, with each character representing a binary
# number. Each character in the string will be '0' or '1'.
# There will be an extra couple of characters 'db' at the beginning and at the end of the string.
# The extra characters are there to help with the format.
# Examples:
# decimal_to_binary(15)   # returns "db1111db"
# decimal_to_binary(32)   # returns "db100000db"
decimal_to_binary <- function(decimal) {
    # your code here
}


test_humaneval <- function() {
candidate <- decimal_to_binary
    if(!identical(candidate(0), 'db0db')){quit('no', 1)}
    if(!identical(candidate(32), 'db100000db')){quit('no', 1)}
    if(!identical(candidate(103), 'db1100111db')){quit('no', 1)}
    if(!identical(candidate(15), 'db1111db')){quit('no', 1)}
}
test_humaneval()